Sage/carbon
# fix(menu-full-screen): ensure no crash when clicking outside a submenu

fix #5682

### Proposed behaviour

A MenuFullScreen component should not crash when clicking outside the menu, or on a non-submenu menu item.



### Current behaviour

The app crashes with an error `setOpenSubmenuIndex is not a function`.



### Checklist




#### QA


### Additional context



### Testing instructions



Test both this codesanbox https://codesandbox.io/s/fe-5578-menu-typeerror-xr66wk
And on Storybook (on a small enough screen size to view the full-screen versions of the menus): ?path=/story/menu--fullscreen-view

Check that no clicks anywhere on the page cause a console error.
